reactjs-films-homework/src/pages/ActorProfile/ActorProfile.tsx

```diff
@@ -11,48 +8,64 @@ import { IMovieCard } from '../../components/MovieCard/MovieCard';
 import { IPhotos } from './ActorPhotos/ActorPhotos';
 import { IActorTitleProps } from './ActorTitleInfo/ActorTitleInfo';
 
+import { useAppSelector, useAppDispatch } from '../../hooks/hooks';
+import getActorInfo from '../../store/rootStore/actorPageStore/getActorData/getActorInfo';
+import getActorImages from '../../store/rootStore/actorPageStore/getActorData/getActorPhotos';
+import getFilmsWithActor from '../../store/rootStore/actorPageStore/getActorData/getFilmsWithActor';
+
 const ActorProfile: React.FC = () => {
   const [info, setInfo] = useState<IActorTitleProps>();
   const [photos, setPhotos] = useState<Array<IPhotos>>([]);
   const [knownBy, setKnownBy] = useState<Array<IMovieCard>>([]);
   const [sortKnownBy, setSortKnownBy] = useState<Array<IMovieCard>>([]);
   const actorGridPhotosLength = 4;
+  const filmsLength = 10;
+  const dispatch = useAppDispatch();
 
-  useEffect(() => {
-    actorInfo.then((res) => {
-      setInfo((): IActorTitleProps => res);
-      return;
-    });
-    actorPhotosMock.then((res) => {
-      setPhotos((): IPhotos[] => res.profiles);
-    });
-    filmsWithActor.then((res) => {
-      setKnownBy((): any[] => res.cast);
-    });
-  });
-
-  const sortFilmPopularity = (arr: Array<IMovieCard>) => {
-    const filmsLength = 10;
-    const resultFilmsLength = [...arr].sort((a, b) => {
-      return a.vote_average > b.vote_average ? -1 : 1;
-    });
-    return resultFilmsLength.slice(0, filmsLength);
+  const actorInfoStore = useAppSelector((state) => state.actorPageReducer);
+  const lang = useAppSelector((state) => state.mainReducer.lang);
+  const id = useAppSelector((state) => state.actorPageReducer.id);
+  const cast = useAppSelector((state) => state.actorPageReducer.cast);
+
+  const sortFilmPopularity = (arr: Array<IMovieCard>, filmsLength: number) => {
+    // const resultFilmsLength = [...arr].sort((a, b) => {
+    //   return a.vote_average > b.vote_average ? -1 : 1;
+    // });
+    // I don't know how it will be better
+    return arr.slice(0, filmsLength);
   };
 
   useMemo(() => {
-    setSortKnownBy(sortFilmPopularity(knownBy));
+    const requestInfoProperties = {
+      lang: lang,
+      id: id,
+    };
+    dispatch(getActorInfo(requestInfoProperties));
+    dispatch(getFilmsWithActor(requestInfoProperties));
+    dispatch(getActorImages(id));
+  }, [lang, id, dispatch]);
```

Comment:
In general fetching data performes in useEffect, while data is fetching a loader should render on the page, after that you can render the component with needed data. There is no need to set this data in the local state by useState hooks. You can just get it directly from the store when data is fetched.